homeassistant/components/sensor/miflora.py

```diff
@@ -0,0 +1,142 @@
+"""
+Support for Xiaomi Mi Flora BLE plant sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.miflora/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.const import CONF_MONITORED_CONDITIONS, CONF_NAME
+
+
+REQUIREMENTS = ['miflora==0.1.4']
+
+LOGGER = logging.getLogger(__name__)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=900)
+CONF_MAC = "mac"
+CONF_FORCE_UPDATE = 'force_update'
+DEFAULT_NAME = ""
+
+# Sensor types are defined like: Name, units
+SENSOR_TYPES = {
+    'temperature': ['Temperature', '°C'],
+    'light': ['Light intensity', 'lux'],
+    'moisture': ['Moisture', '%'],
+    'conductivity': ['Conductivity', 'µS/cm'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MAC): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_FORCE_UPDATE, default=False): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the MiFlora sensor."""
+    from miflora import miflora_poller
+
+    poller = miflora_poller.MiFloraPoller(config.get(CONF_MAC))
```

Comment:
BLE devices support only a single connection. Therefore it is always possible that a lookup fails. Making the platform fail due to a temporary problem will leave the user without a sensor that might be available a few seconds later again. I find this much better than removing a sensor just because it is not available during startup.